fastly/block_fastly_service_v1_gzip_test.go

```diff
@@ -272,3 +291,32 @@ resource "fastly_service_v1" "foo" {
   force_destroy = true
 }`, name, domain)
 }
+
+func testAccServiceV1GzipsConfig_update(name, domain string) string {
+	return fmt.Sprintf(`
+resource "fastly_service_v1" "foo" {
+  name = "%s"
+
+  domain {
+    name    = "%s"
+    comment = "tf-testing-domain"
+  }
+
+  backend {
+    address = "aws.amazon.com"
+    name    = "amazon docs"
+  }
+
+  gzip {
```

Comment:
I want to clarify why this new test validates the actual 'update' mechanics...

The original test _thought_ it was validating a 'modification' to the gzip resource, but because they changed the _name_ field (along with other fields) it meant that the gzip resource was actually being 'deleted' and then 'recreated' so the 'modification' logic (which is where the panic happens) was never actually run.

To ensure the modification/update logic is executed I created a new test that updated the gzip resource but left the name the same and just modified some of the other fields.